src/turbomind/models/llama/SequenceManager.cc

```diff
@@ -226,6 +250,13 @@ struct Transaction {
             count -= tmp;
             evict_ += tmp;
 
+            // evict from block trie
+            if (block_trie_->enabled()) {
+                cache_evict_ = block_trie_->evict(count);
```

Comment:
I know here just do the planning not real execution. But I still evict here due to the following points:
1. It's hard to get the actual evictable number from the block trie because we evict from the dynamic leaves.
2. It's fine to evict blocks from block trie here, because we have already achieved the goal: free the cached blocks lazily.
3. The blocks will not be free immediately here because we just decrease the ref count of these blocks in the block trie eviction. They will in cached status and will be freed when `block_manager_->Evict(schedule.evict)` is executed.

Reply:
We hope keep the blocks in the trie cache as long as possible. But evicting some blocks by mistake (transactions not committed) is still acceptable here. These blocks will be put into the cached pool (ref_count==0) and can be allocated in the next iteration.

Reply:
From my perspective cache blocks should be managed by existing mechanism instead of by block trie. Block trie needs to verify the validity of the blocks as the sequence manager does.

Reply:
In current implementation, the block trie holds one `use_count` for the blocks in it to mark the cached blocks. 
If use existing mechanism to manage the cache blocks, do you mean the block trie doesn't need to hold the `use_count`, and before matching, we should traverse the block trie to remove the invalid nodes which block unique_id not matched?

Reply:
> do you mean the block trie doesn't need to hold the `use_count`, and before matching, we should traverse the block trie to remove the invalid nodes which block unique_id not matched?

Yes.

Reply:
@lzhangzz It's better to evict the blocks in block trie from the leaves. If use the existing cache mechanism, we cannot make sure the evicted block is leaf in the block trie. If one block is not the leaf, then all the children will be dropped.

Reply:
With the reversed timestamping order of the blocks in `Touch` , it's guaranteed to evict from the leaves.

Reply:
@lzhangzz The cache logic is updated, pls help review again. Thanks!